src/private/types/policy_store_id.rs

```diff
@@ -2,10 +2,12 @@
 
 use std::fmt;
 
+use super::policy_store_filter::PolicyStoreFilters;
+
 /// This Object wraps the aws verified permissions `PolicyStoreID` which is an unique identifier
 /// for the policy store.
-#[derive(Debug, Clone, Eq, PartialEq, Hash)]
-pub struct PolicyStoreId(pub String);
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PolicyStoreId(pub String, pub Option<PolicyStoreFilters>);
```

Comment:
> I'd prefer the filter to be a separate type rather than part of the `PolicyStoreId`. Is the motivation for this to prevent breaking changes? We're actually planning breaking changes to this crate in the near term for upgrading to Cedar 4, so this can be rolled in with that if you need to make breaking changes.

The reason the filter is part of the `PolicyStoreId` is that it forms part of the key to the cache - that is, each cache _can_ now hold just a _subset_ of the policies in an AVP Policy Store, filtered by the provided filter. In order to correctly identify the cache, then, we need the filter to be part of the key to the cache HashMap. 

Another way to do this would be, of course, to have a `HashMap<PolicyStoreId, HashMap<Option<PolicyFilter>, T>>` but that seemed to me to be a little obfuscated, adding unnecessary complexity, and a slight performance degradation.